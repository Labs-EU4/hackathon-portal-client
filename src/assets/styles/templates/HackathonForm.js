import styled from 'styled-components';
import { Form } from "formik";

import Button from "../../../components/atoms/Button";

export const FormContainer = styled.div`
  position: relative;
  width: 100%; height: 100%;
  padding-top: 20px;
`;

export const StyledForm = styled(Form)`
  display: flex;
  padding: 20px;
`;

export const StyledColumn = styled.div`
  width: 50%; min-height: 70vh;
  padding: 0 10px;

  &:first-child {
    border-right: 2px solid ${props => props.theme.color.grey.border};
  }
`;

export const StyledFormBtn = styled(Button)`
  position: absolute; top: calc(100% - 100px); left: 50%;
  width: 50%;
  margin-top: 30px;
  transform: translateX(-50%);
`;