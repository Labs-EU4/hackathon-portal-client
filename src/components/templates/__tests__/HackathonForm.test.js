import React from "react";
import { Router } from "react-router";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { render, fireEvent, cleanup } from "@testing-library/react";
import configureStore from "redux-mock-store";
import "@testing-library/jest-dom/extend-expect";
import HackathonForm from "../HackathonForm";
import { initialState } from "../../../utils/mockData";
import { theme } from "../../../assets/styles/ThemeStyling";
import { ThemeProvider } from "styled-components";
import { library } from "@fortawesome/fontawesome-svg-core";
import { fas } from "@fortawesome/free-solid-svg-icons";
import { far } from "@fortawesome/free-regular-svg-icons";
import { fab } from "@fortawesome/free-brands-svg-icons";

library.add(fas, far, fab);

const history = createMemoryHistory();

afterEach(cleanup);

let jestFeatures;
let mockStore;
let store;

beforeEach(() => {
  mockStore = configureStore();
  store = mockStore(initialState);
  const event = initialState.events.data[0];
  jestFeatures = render(
    <Router history={history}>
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <HackathonForm event={event} />
        </ThemeProvider>
      </Provider>
    </Router>
  );
});

describe("Shows all the text nodes on HackathonForm.js that are contained on the making an event form", () => {
  it("should be displaying the label text node for the form ", () => {
    let mainHeader = () => jestFeatures.getByText("Hackathon Title");
    expect(mainHeader()).toBeInTheDocument();
  });
  it("should be displaying the label text node for the start date", () => {
    expect(jestFeatures.getByText("Event Starts")).toBeInTheDocument();
  });
  it("should be displaying the label text node for the end date", () => {
    expect(jestFeatures.getByText("Event Ends")).toBeInTheDocument();
  });
  it("should be displaying the description text node for the event being created", () => {
    expect(jestFeatures.getByText("Description")).toBeInTheDocument();
  });
  it("should be displaying the tags text node for the event being created", () => {
    expect(jestFeatures.getByText("Tags")).toBeInTheDocument();
  });
  it("should be displaying the Participation type text node for the event being created", () => {
    expect(jestFeatures.getByText("Participation Type")).toBeInTheDocument();
  });
  it("should be displaying the option text node for the event being created", () => {
    expect(jestFeatures.getByText("Individual")).toBeInTheDocument();
  });
  it("should be displaying the category text node for the event being created", () => {
    expect(jestFeatures.getByText("Event Category")).toBeInTheDocument();
  });
  it("should be displaying the location text node for the event being created", () => {
    expect(jestFeatures.getByText("Location")).toBeInTheDocument();
  });
  it("should be displaying the category name text node for the event being created", () => {
    expect(jestFeatures.getByText("Summer Hackton")).toBeInTheDocument();
  });
  it("should be displaying the fifth category name text node for the event being created", () => {
    expect(jestFeatures.getByText("Advanced Hackton")).toBeInTheDocument();
  });

  it("should be displaying the 'Create New Hackathon' text node for the event being created", () => {
    expect(jestFeatures.getByText(/Create New Hackathon/i)).toBeInTheDocument();
  });
  it("should be displaying the grading rubric's description text node for the event being created", () => {
    expect(
      jestFeatures.getByText(
        /Judges will be expected to grade project submissions on which one of the following/i
      )
    ).toBeInTheDocument();
  });

  it("should be displaying the guidelines text node for the event being created", () => {
    expect(jestFeatures.getByText("Guidelines")).toBeInTheDocument();
  });

  it("should be displaying 'Judges will be expected' text node for the event being created", () => {
    expect(
      jestFeatures.getByText(/Judges will be expected/i)
    ).toBeInTheDocument();
  });

  it("should be displaying the paragraph project requirements text node for the event being created", () => {
    expect(
      jestFeatures.getByText(
        /Participants will be expected to submit which one of the following/i
      )
    ).toBeInTheDocument();
  });
  it("should be displaying 'Event Starts' text node for the event being created", () => {
    expect(jestFeatures.getByText(/Event Starts/i)).toBeInTheDocument();
  });

  it("renders without crashing ", () => {
    expect(jestFeatures).toMatchSnapshot();
  });
});

describe("Shows that all the rubrics checkbox values are being displayed properly", () => {
  it("should be displaying the first rubric value checkbox for the event being created", () => {
    expect(jestFeatures.getByDisplayValue("presentation")).toBeInTheDocument();
  });
  it("should be displaying the second rubric value checkbox for the event being created", () => {
    expect(jestFeatures.getByDisplayValue("innovation")).toBeInTheDocument();
  });
  it("should be displaying the third rubric value checkbox for the event being created", () => {
    expect(jestFeatures.getByDisplayValue("product_fit")).toBeInTheDocument();
  });
  it("should be displaying the fourth rubric value checkbox for the event being created", () => {
    expect(
      jestFeatures.getByDisplayValue("product_design")
    ).toBeInTheDocument();
  });
  it("should be displaying the fifth rubric value checkbox for the event being created", () => {
    expect(jestFeatures.getByDisplayValue("functionality")).toBeInTheDocument();
  });
  it("should be displaying the sixth rubric value checkbox for the event being created", () => {
    expect(jestFeatures.getByDisplayValue("extensibility")).toBeInTheDocument();
  });
  it("should be displaying the video rubric value checkbox for the event being created", () => {
    expect(jestFeatures.getByDisplayValue("video_url")).toBeInTheDocument();
  });
  it("should be displaying the github rubric value checkbox for the event being created", () => {
    expect(jestFeatures.getByDisplayValue("github_url")).toBeInTheDocument();
  });
});

describe("Checks for the button on the UI", () => {
  it("should be displaying submit button", () => {
    const submit = jestFeatures.queryAllByText(/Submit/i)[0];
    expect(submit).toBeDefined();
  });
});
