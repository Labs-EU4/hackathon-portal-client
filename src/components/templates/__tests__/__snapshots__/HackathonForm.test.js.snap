// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows all the text nodes on HackathonForm.js that are contained on the making an event form renders without crashing  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="sc-bdVaJa gOhKpA"
        >
          <div
            class="sc-htpNat dAVxPZ"
          >
            <a
              href="/dashboard"
            >
              <img
                alt="Hackton - Organise hackathons"
                class="sc-ifAKCX gCNDrT"
                src="Hackton-logo.png"
              />
            </a>
            <div
              class="sc-iQNlJl jfTdcZ"
            >
              <div
                class="sc-EHOje kXozTT"
              >
                8
                <ul
                  class="sc-bZQynM fVGedF"
                >
                  <li>
                    <a
                      href="/dashboard/profile/edit"
                    >
                      Settings
                    </a>
                  </li>
                  <li>
                    <span>
                      LogOut
                    </span>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-jhAzac jsMUOz"
        >
          <div
            class="sc-fMiknA sc-fYxtnH fViRow"
          >
            <div
              class="sc-fAjcbJ kUlDMA"
            >
              <h3
                class="sc-eNQAEJ faMTgL"
              >
                Create New Hackathon
              </h3>
            </div>
            <div
              class="sc-cHGsZl cEmgZG"
            >
              <div
                class="sc-kgAjT sc-ksYbfQ dnkmpe"
              >
                <form
                  action="#"
                >
                  <div
                    class="sc-gisBJw dTDHDQ"
                  >
                    <label
                      class="sc-hqyNC gKOPvS"
                      for="event_title"
                    >
                      Hackathon Title
                    </label>
                    <input
                      class="sc-eHgmQL cwnFTa"
                      display="wide"
                      id="event_title"
                      name="event_title"
                      type="text"
                      value=""
                    />
                    <span
                      class="sc-brqgnP sc-jDwBTQ erbhIA"
                    />
                  </div>
                  <div
                    class="sc-gisBJw dTDHDQ"
                  >
                    <div
                      class="sc-cHGsZl cEmgZG"
                    >
                      <label
                        class="sc-hqyNC gKOPvS"
                        for="start_date"
                      >
                        Event Starts
                      </label>
                      <input
                        class="sc-eHgmQL eUgCMc"
                        id="start_date"
                        min="2020-03-16"
                        name="start_date"
                        placeholder="Event starts"
                        type="date"
                        value="2020-03-16"
                      />
                      <span
                        class="sc-brqgnP sc-jDwBTQ erbhIA"
                      />
                    </div>
                    <div
                      class="sc-cHGsZl cEmgZG"
                    >
                      <label
                        class="sc-hqyNC gKOPvS"
                        for="end_date"
                      >
                        Event Ends
                      </label>
                      <input
                        class="sc-eHgmQL eUgCMc"
                        id="end_date"
                        min=""
                        name="end_date"
                        placeholder="Event ends"
                        type="date"
                        value="2020-03-16"
                      />
                      <span
                        class="sc-brqgnP sc-jDwBTQ erbhIA"
                      />
                    </div>
                  </div>
                  <div
                    class="sc-gisBJw dTDHDQ"
                  >
                    <label
                      class="sc-hqyNC gKOPvS"
                      for="event_description"
                    >
                      Description
                    </label>
                    <div
                      class="sc-gGBfsJ iilUTs"
                    >
                      <textarea
                        id="event_description"
                        name="event_description"
                        type="text"
                      />
                    </div>
                    <span
                      class="sc-brqgnP sc-jDwBTQ erbhIA"
                    />
                  </div>
                  <div
                    class="sc-gisBJw dTDHDQ"
                  >
                     
                    <label
                      class="sc-hqyNC gKOPvS"
                      for="input_tags"
                    >
                      Tags
                    </label>
                    <div
                      class="input-tag"
                      id="input_tags"
                      tags=""
                    >
                      <ul
                        class="input-tag__tags"
                      >
                        <li
                          class="input-tag__tags__input"
                        >
                          <input
                            maxlength="20"
                            type="text"
                          />
                        </li>
                      </ul>
                    </div>
                  </div>
                  <div
                    class="sc-gisBJw dTDHDQ"
                  >
                    <div
                      class="sc-cHGsZl cEmgZG"
                    >
                      <label
                        class="sc-hqyNC gKOPvS"
                        for="participation_type"
                      >
                        Participation Type
                      </label>
                      <select
                        class="sc-jnlKLf jSfQSO"
                        id="participation_type"
                        name="participation_type"
                      >
                        <option
                          disabled=""
                          hidden=""
                          value=""
                        >
                          Choose
                        </option>
                        <option
                          value="individual"
                        >
                          Individual
                        </option>
                        <option
                          value="team"
                        >
                          Team
                        </option>
                        <option
                          value="both"
                        >
                          Both
                        </option>
                      </select>
                      <span
                        class="sc-brqgnP sc-jDwBTQ erbhIA"
                      />
                    </div>
                    <div
                      class="sc-cHGsZl cEmgZG"
                    >
                      <label
                        class="sc-hqyNC gKOPvS"
                        for="event_category"
                      >
                        Event Category
                      </label>
                      <select
                        class="sc-jnlKLf jSfQSO"
                        id="event_category"
                        name="event_category"
                      >
                        <option
                          value=""
                        >
                          Choose
                        </option>
                        <option
                          value="2"
                        >
                          Summer Hackton
                        </option>
                        <option
                          value="3"
                        >
                          World Hackton
                        </option>
                        <option
                          value="3"
                        >
                          Asia Hackton
                        </option>
                        <option
                          value="3"
                        >
                          Gamer's Hackton
                        </option>
                        <option
                          value="3"
                        >
                          Advanced Hackton
                        </option>
                      </select>
                      <span
                        class="sc-brqgnP sc-jDwBTQ erbhIA"
                      />
                    </div>
                  </div>
                  <div
                    class="sc-gisBJw dTDHDQ"
                  >
                    <label
                      class="sc-hqyNC gKOPvS"
                      for="location"
                    >
                      Location
                    </label>
                    <input
                      class="sc-eHgmQL cwnFTa"
                      display="wide"
                      id="location"
                      name="location"
                      type="text"
                      value=""
                    />
                    <span
                      class="sc-brqgnP sc-jDwBTQ erbhIA"
                    />
                  </div>
                  <div
                    class="sc-gisBJw dTDHDQ"
                    id="grading_rubrics"
                  >
                    <label
                      class="sc-hqyNC gKOPvS"
                      for="grading_rubrics"
                    >
                      Grading Rubrics
                    </label>
                    <p
                      class="sc-kAzzGY brIVdx"
                    >
                      Judges will be expected to grade project submissions on which one of the following* (tick on all that apply)
                    </p>
                    <div
                      class="sc-jqCOkK hRLaDQ"
                    >
                      <input
                        class="sc-bbmXgH hsTpOe"
                        id="presentation"
                        name="rubrics"
                        type="checkbox"
                        value="presentation"
                      />
                      <label
                        class="sc-jbKcbu fRzHzg"
                        for="presentation"
                      >
                        Presentation
                      </label>
                    </div>
                    <div
                      class="sc-jqCOkK hRLaDQ"
                    >
                      <input
                        class="sc-bbmXgH hsTpOe"
                        id="product_fit"
                        name="rubrics"
                        type="checkbox"
                        value="product_fit"
                      />
                      <label
                        class="sc-jbKcbu fRzHzg"
                        for="product_fit"
                      >
                        Product Fit
                      </label>
                    </div>
                    <div
                      class="sc-jqCOkK hRLaDQ"
                    >
                      <input
                        class="sc-bbmXgH hsTpOe"
                        id="functionality"
                        name="rubrics"
                        type="checkbox"
                        value="functionality"
                      />
                      <label
                        class="sc-jbKcbu fRzHzg"
                        for="functionality"
                      >
                        Functionality
                      </label>
                    </div>
                    <div
                      class="sc-jqCOkK hRLaDQ"
                    >
                      <input
                        class="sc-bbmXgH hsTpOe"
                        id="innovation"
                        name="rubrics"
                        type="checkbox"
                        value="innovation"
                      />
                      <label
                        class="sc-jbKcbu fRzHzg"
                        for="innovation"
                      >
                        Innovation
                      </label>
                    </div>
                    <div
                      class="sc-jqCOkK hRLaDQ"
                    >
                      <input
                        class="sc-bbmXgH hsTpOe"
                        id="product_design"
                        name="rubrics"
                        type="checkbox"
                        value="product_design"
                      />
                      <label
                        class="sc-jbKcbu fRzHzg"
                        for="product_design"
                      >
                        Product Design
                      </label>
                    </div>
                    <div
                      class="sc-jqCOkK hRLaDQ"
                    >
                      <input
                        class="sc-bbmXgH hsTpOe"
                        id="extensibility"
                        name="rubrics"
                        type="checkbox"
                        value="extensibility"
                      />
                      <label
                        class="sc-jbKcbu fRzHzg"
                        for="extensibility"
                      >
                        Extensibility
                      </label>
                    </div>
                    <span
                      class="sc-brqgnP sc-jDwBTQ erbhIA"
                    />
                  </div>
                  <div
                    class="sc-gisBJw dTDHDQ"
                  >
                    <label
                      class="sc-hqyNC gKOPvS"
                      for="guidelines"
                    >
                      Guidelines
                    </label>
                    <div
                      class="sc-gGBfsJ iilUTs"
                    >
                      <textarea
                        id="guidelines"
                        name="guidelines"
                        type="text"
                      />
                    </div>
                    <span
                      class="sc-brqgnP sc-jDwBTQ erbhIA"
                    />
                  </div>
                  <label
                    class="sc-hqyNC gKOPvS"
                    for="submission_requirements"
                  >
                    Project Submission Requirements
                  </label>
                  <div
                    class="sc-gisBJw dTDHDQ"
                    id="submission_requirements"
                  >
                    <p
                      class="sc-kAzzGY brIVdx"
                    >
                      Participants will be expected to submit which one of the following (tick on all that apply)
                    </p>
                    <div
                      class="sc-jqCOkK hRLaDQ"
                    >
                      <input
                        class="sc-bbmXgH hsTpOe"
                        id="video_url"
                        name="requirements"
                        type="checkbox"
                        value="video_url"
                      />
                      <label
                        class="sc-jbKcbu fRzHzg"
                        for="video_url"
                      >
                        Video URL
                      </label>
                    </div>
                    <div
                      class="sc-jqCOkK hRLaDQ"
                    >
                      <input
                        class="sc-bbmXgH hsTpOe"
                        id="github_url"
                        name="requirements"
                        type="checkbox"
                        value="github_url"
                      />
                      <label
                        class="sc-jbKcbu fRzHzg"
                        for="github_url"
                      >
                        GitHub URL
                      </label>
                    </div>
                    <span
                      class="sc-brqgnP sc-jDwBTQ erbhIA"
                    />
                  </div>
                  <div
                    class="sc-gisBJw dTDHDQ"
                  >
                    <div
                      class="sc-epnACN bHPajQ"
                    >
                      <a
                        class="sc-jWBwVP hQvxVm"
                        color="grey"
                        href="/dashboard"
                      >
                        Cancel
                      </a>
                      <button
                        class="sc-cvbbAY cUQDWK"
                        color="green"
                        type="submit"
                      >
                        Submit
                      </button>
                    </div>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-gzVnrw bRGEaa"
        >
          <div
            class="sc-dnqmqq dBtDWt"
          >
            <div>
              <img
                alt="LinkedIn"
                class="sc-gZMcBi fKNoMs"
                src="Icon-linkedin.png"
              />
              <img
                alt="Twitter"
                class="sc-gZMcBi fKNoMs"
                src="Icon-twitter.png"
              />
              <img
                alt="Facebook"
                class="sc-gZMcBi fKNoMs"
                src="Icon-facebook.png"
              />
            </div>
            <div>
              <a
                class="sc-gqjmRU sc-jTzLTM fddzAd"
                href="https://hackton.co/#team"
              >
                Team
              </a>
              <a
                class="sc-gqjmRU sc-jTzLTM fddzAd"
                href="#"
              >
                Privacy
              </a>
              <a
                class="sc-gqjmRU sc-jTzLTM fddzAd"
                href="#"
              >
                Contacts
              </a>
            </div>
            <div>
              <p
                class="sc-kAzzGY sc-chPdSV gAkGJE"
              >
                International Crafters © 2020
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="sc-bdVaJa gOhKpA"
      >
        <div
          class="sc-htpNat dAVxPZ"
        >
          <a
            href="/dashboard"
          >
            <img
              alt="Hackton - Organise hackathons"
              class="sc-ifAKCX gCNDrT"
              src="Hackton-logo.png"
            />
          </a>
          <div
            class="sc-iQNlJl jfTdcZ"
          >
            <div
              class="sc-EHOje kXozTT"
            >
              8
              <ul
                class="sc-bZQynM fVGedF"
              >
                <li>
                  <a
                    href="/dashboard/profile/edit"
                  >
                    Settings
                  </a>
                </li>
                <li>
                  <span>
                    LogOut
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-jhAzac jsMUOz"
      >
        <div
          class="sc-fMiknA sc-fYxtnH fViRow"
        >
          <div
            class="sc-fAjcbJ kUlDMA"
          >
            <h3
              class="sc-eNQAEJ faMTgL"
            >
              Create New Hackathon
            </h3>
          </div>
          <div
            class="sc-cHGsZl cEmgZG"
          >
            <div
              class="sc-kgAjT sc-ksYbfQ dnkmpe"
            >
              <form
                action="#"
              >
                <div
                  class="sc-gisBJw dTDHDQ"
                >
                  <label
                    class="sc-hqyNC gKOPvS"
                    for="event_title"
                  >
                    Hackathon Title
                  </label>
                  <input
                    class="sc-eHgmQL cwnFTa"
                    display="wide"
                    id="event_title"
                    name="event_title"
                    type="text"
                    value=""
                  />
                  <span
                    class="sc-brqgnP sc-jDwBTQ erbhIA"
                  />
                </div>
                <div
                  class="sc-gisBJw dTDHDQ"
                >
                  <div
                    class="sc-cHGsZl cEmgZG"
                  >
                    <label
                      class="sc-hqyNC gKOPvS"
                      for="start_date"
                    >
                      Event Starts
                    </label>
                    <input
                      class="sc-eHgmQL eUgCMc"
                      id="start_date"
                      min="2020-03-16"
                      name="start_date"
                      placeholder="Event starts"
                      type="date"
                      value="2020-03-16"
                    />
                    <span
                      class="sc-brqgnP sc-jDwBTQ erbhIA"
                    />
                  </div>
                  <div
                    class="sc-cHGsZl cEmgZG"
                  >
                    <label
                      class="sc-hqyNC gKOPvS"
                      for="end_date"
                    >
                      Event Ends
                    </label>
                    <input
                      class="sc-eHgmQL eUgCMc"
                      id="end_date"
                      min=""
                      name="end_date"
                      placeholder="Event ends"
                      type="date"
                      value="2020-03-16"
                    />
                    <span
                      class="sc-brqgnP sc-jDwBTQ erbhIA"
                    />
                  </div>
                </div>
                <div
                  class="sc-gisBJw dTDHDQ"
                >
                  <label
                    class="sc-hqyNC gKOPvS"
                    for="event_description"
                  >
                    Description
                  </label>
                  <div
                    class="sc-gGBfsJ iilUTs"
                  >
                    <textarea
                      id="event_description"
                      name="event_description"
                      type="text"
                    />
                  </div>
                  <span
                    class="sc-brqgnP sc-jDwBTQ erbhIA"
                  />
                </div>
                <div
                  class="sc-gisBJw dTDHDQ"
                >
                   
                  <label
                    class="sc-hqyNC gKOPvS"
                    for="input_tags"
                  >
                    Tags
                  </label>
                  <div
                    class="input-tag"
                    id="input_tags"
                    tags=""
                  >
                    <ul
                      class="input-tag__tags"
                    >
                      <li
                        class="input-tag__tags__input"
                      >
                        <input
                          maxlength="20"
                          type="text"
                        />
                      </li>
                    </ul>
                  </div>
                </div>
                <div
                  class="sc-gisBJw dTDHDQ"
                >
                  <div
                    class="sc-cHGsZl cEmgZG"
                  >
                    <label
                      class="sc-hqyNC gKOPvS"
                      for="participation_type"
                    >
                      Participation Type
                    </label>
                    <select
                      class="sc-jnlKLf jSfQSO"
                      id="participation_type"
                      name="participation_type"
                    >
                      <option
                        disabled=""
                        hidden=""
                        value=""
                      >
                        Choose
                      </option>
                      <option
                        value="individual"
                      >
                        Individual
                      </option>
                      <option
                        value="team"
                      >
                        Team
                      </option>
                      <option
                        value="both"
                      >
                        Both
                      </option>
                    </select>
                    <span
                      class="sc-brqgnP sc-jDwBTQ erbhIA"
                    />
                  </div>
                  <div
                    class="sc-cHGsZl cEmgZG"
                  >
                    <label
                      class="sc-hqyNC gKOPvS"
                      for="event_category"
                    >
                      Event Category
                    </label>
                    <select
                      class="sc-jnlKLf jSfQSO"
                      id="event_category"
                      name="event_category"
                    >
                      <option
                        value=""
                      >
                        Choose
                      </option>
                      <option
                        value="2"
                      >
                        Summer Hackton
                      </option>
                      <option
                        value="3"
                      >
                        World Hackton
                      </option>
                      <option
                        value="3"
                      >
                        Asia Hackton
                      </option>
                      <option
                        value="3"
                      >
                        Gamer's Hackton
                      </option>
                      <option
                        value="3"
                      >
                        Advanced Hackton
                      </option>
                    </select>
                    <span
                      class="sc-brqgnP sc-jDwBTQ erbhIA"
                    />
                  </div>
                </div>
                <div
                  class="sc-gisBJw dTDHDQ"
                >
                  <label
                    class="sc-hqyNC gKOPvS"
                    for="location"
                  >
                    Location
                  </label>
                  <input
                    class="sc-eHgmQL cwnFTa"
                    display="wide"
                    id="location"
                    name="location"
                    type="text"
                    value=""
                  />
                  <span
                    class="sc-brqgnP sc-jDwBTQ erbhIA"
                  />
                </div>
                <div
                  class="sc-gisBJw dTDHDQ"
                  id="grading_rubrics"
                >
                  <label
                    class="sc-hqyNC gKOPvS"
                    for="grading_rubrics"
                  >
                    Grading Rubrics
                  </label>
                  <p
                    class="sc-kAzzGY brIVdx"
                  >
                    Judges will be expected to grade project submissions on which one of the following* (tick on all that apply)
                  </p>
                  <div
                    class="sc-jqCOkK hRLaDQ"
                  >
                    <input
                      class="sc-bbmXgH hsTpOe"
                      id="presentation"
                      name="rubrics"
                      type="checkbox"
                      value="presentation"
                    />
                    <label
                      class="sc-jbKcbu fRzHzg"
                      for="presentation"
                    >
                      Presentation
                    </label>
                  </div>
                  <div
                    class="sc-jqCOkK hRLaDQ"
                  >
                    <input
                      class="sc-bbmXgH hsTpOe"
                      id="product_fit"
                      name="rubrics"
                      type="checkbox"
                      value="product_fit"
                    />
                    <label
                      class="sc-jbKcbu fRzHzg"
                      for="product_fit"
                    >
                      Product Fit
                    </label>
                  </div>
                  <div
                    class="sc-jqCOkK hRLaDQ"
                  >
                    <input
                      class="sc-bbmXgH hsTpOe"
                      id="functionality"
                      name="rubrics"
                      type="checkbox"
                      value="functionality"
                    />
                    <label
                      class="sc-jbKcbu fRzHzg"
                      for="functionality"
                    >
                      Functionality
                    </label>
                  </div>
                  <div
                    class="sc-jqCOkK hRLaDQ"
                  >
                    <input
                      class="sc-bbmXgH hsTpOe"
                      id="innovation"
                      name="rubrics"
                      type="checkbox"
                      value="innovation"
                    />
                    <label
                      class="sc-jbKcbu fRzHzg"
                      for="innovation"
                    >
                      Innovation
                    </label>
                  </div>
                  <div
                    class="sc-jqCOkK hRLaDQ"
                  >
                    <input
                      class="sc-bbmXgH hsTpOe"
                      id="product_design"
                      name="rubrics"
                      type="checkbox"
                      value="product_design"
                    />
                    <label
                      class="sc-jbKcbu fRzHzg"
                      for="product_design"
                    >
                      Product Design
                    </label>
                  </div>
                  <div
                    class="sc-jqCOkK hRLaDQ"
                  >
                    <input
                      class="sc-bbmXgH hsTpOe"
                      id="extensibility"
                      name="rubrics"
                      type="checkbox"
                      value="extensibility"
                    />
                    <label
                      class="sc-jbKcbu fRzHzg"
                      for="extensibility"
                    >
                      Extensibility
                    </label>
                  </div>
                  <span
                    class="sc-brqgnP sc-jDwBTQ erbhIA"
                  />
                </div>
                <div
                  class="sc-gisBJw dTDHDQ"
                >
                  <label
                    class="sc-hqyNC gKOPvS"
                    for="guidelines"
                  >
                    Guidelines
                  </label>
                  <div
                    class="sc-gGBfsJ iilUTs"
                  >
                    <textarea
                      id="guidelines"
                      name="guidelines"
                      type="text"
                    />
                  </div>
                  <span
                    class="sc-brqgnP sc-jDwBTQ erbhIA"
                  />
                </div>
                <label
                  class="sc-hqyNC gKOPvS"
                  for="submission_requirements"
                >
                  Project Submission Requirements
                </label>
                <div
                  class="sc-gisBJw dTDHDQ"
                  id="submission_requirements"
                >
                  <p
                    class="sc-kAzzGY brIVdx"
                  >
                    Participants will be expected to submit which one of the following (tick on all that apply)
                  </p>
                  <div
                    class="sc-jqCOkK hRLaDQ"
                  >
                    <input
                      class="sc-bbmXgH hsTpOe"
                      id="video_url"
                      name="requirements"
                      type="checkbox"
                      value="video_url"
                    />
                    <label
                      class="sc-jbKcbu fRzHzg"
                      for="video_url"
                    >
                      Video URL
                    </label>
                  </div>
                  <div
                    class="sc-jqCOkK hRLaDQ"
                  >
                    <input
                      class="sc-bbmXgH hsTpOe"
                      id="github_url"
                      name="requirements"
                      type="checkbox"
                      value="github_url"
                    />
                    <label
                      class="sc-jbKcbu fRzHzg"
                      for="github_url"
                    >
                      GitHub URL
                    </label>
                  </div>
                  <span
                    class="sc-brqgnP sc-jDwBTQ erbhIA"
                  />
                </div>
                <div
                  class="sc-gisBJw dTDHDQ"
                >
                  <div
                    class="sc-epnACN bHPajQ"
                  >
                    <a
                      class="sc-jWBwVP hQvxVm"
                      color="grey"
                      href="/dashboard"
                    >
                      Cancel
                    </a>
                    <button
                      class="sc-cvbbAY cUQDWK"
                      color="green"
                      type="submit"
                    >
                      Submit
                    </button>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-gzVnrw bRGEaa"
      >
        <div
          class="sc-dnqmqq dBtDWt"
        >
          <div>
            <img
              alt="LinkedIn"
              class="sc-gZMcBi fKNoMs"
              src="Icon-linkedin.png"
            />
            <img
              alt="Twitter"
              class="sc-gZMcBi fKNoMs"
              src="Icon-twitter.png"
            />
            <img
              alt="Facebook"
              class="sc-gZMcBi fKNoMs"
              src="Icon-facebook.png"
            />
          </div>
          <div>
            <a
              class="sc-gqjmRU sc-jTzLTM fddzAd"
              href="https://hackton.co/#team"
            >
              Team
            </a>
            <a
              class="sc-gqjmRU sc-jTzLTM fddzAd"
              href="#"
            >
              Privacy
            </a>
            <a
              class="sc-gqjmRU sc-jTzLTM fddzAd"
              href="#"
            >
              Contacts
            </a>
          </div>
          <div>
            <p
              class="sc-kAzzGY sc-chPdSV gAkGJE"
            >
              International Crafters © 2020
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
